pkg/controller/common/reconciler/secret.go

```diff
@@ -189,14 +190,14 @@ func GarbageCollectAllSoftOwnedOrphanSecrets(ctx context.Context, c k8s.Client,
 	); err != nil {
 		return err
 	}
-	// remove any secret whose owner in the same namespace doesn't exist
+	// remove any secret whose owner doesn't exist
 	for i := range secrets.Items {
 		secret := secrets.Items[i]
 		softOwner, referenced := SoftOwnerRefFromLabels(secret.Labels)
 		if !referenced {
 			continue
 		}
-		if softOwner.Namespace != secret.Namespace {
+		if restrictedToOwnerNamespace(softOwner.Kind) && softOwner.Namespace != secret.Namespace {
```

Comment:
Since the above changes are for new behavior, shouldn't the unit tests have additional tests covering this new behavior?

Reply:
Yes, good reflex, thank you.